### PR TITLE
[WIP] Adding container image scanning using Trivy

### PR DESCRIPTION
**Pull request by jeremychoi**
Adding a function for container image scanning using Trivy.

Specifying `scanners.trivy.image.name`, a user can scan the image to check if known vulnerabilities exists. See the config-template-long.yaml for more options.

- currently only 'podman' mode is implemented. (supporting 'none' type along with updating Containerfile and build an image will follow later)
- disabled two pylint options

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -43,5 +45,5 @@ repos:
           - --max-line-length=120
           - --min-public-methods=0
           - --good-names=q,f,fp,i,e
-          - --disable=E0401,W1201,W1203,W0603,C0114,C0115,C0116,C0411
+          - --disable=E0401,W1201,W1203,W0603,C0114,C0115,C0116,C0411,W0107,C0209
```
Comment (cedricbu) [inline .pre-commit-config.yaml:48]: should we move this into its own PR ? just to keep purpose separated ?

File: scanners/trivy/trivy.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+        # skip-db-update
+        if self.config.get("scanners.trivy.miscOptions.skipDbUpdate"):
+            cli_ops.extend(["--skip-db-update"])
+
+        image_conf = self.config.get("scanners.trivy.image")
+        if image_conf:
+            if self.config.get("general.container.type") == "podman":
+                # podman run <trivy_image> doesn't require 'trivy' command.
+                self.trivy_cli = ["image"]
```
Comment (cedricbu) [inline scanners/trivy/trivy.py:107]: Ideally, this should be moved to trivy_podman somehow. 
Either in `_setup_podman_cli` or maybe using similar trick as you did in the __init__() with `Setup defaults specific to Podman`, for example storing that into `scanners.trivy.container.parameters.executable` [we already used this entry for ZAP, except that here we will need a list since it can contain 2 elements] => and in this _setup_trivy_cli(), simply `get()` it back

File: scanners/trivy/trivy.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            else:
+                self.trivy_cli = ["trivy image"]
+
+            self.trivy_cli.extend(cli_ops)
+
+            # scanners
+            self.trivy_cli.extend(["--scanners=vuln"])
```
Comment (cedricbu) [inline scanners/trivy/trivy.py:114]: nitpick: why not using `.append()` when adding only 1 element, instead of .extend() + generating a list

File: scanners/trivy/trivy_podman.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+CLASSNAME = "TrivyPodman"
+
+
+pp = pprint.PrettyPrinter(indent=4)
+
+
+class TrivyPodman(Trivy):
+    ###############################################################
+    # PRIVATE CONSTANTS                                           #
+    # Accessed by TrivyPodman only                                  #
```
Comment (cedricbu) [inline scanners/trivy/trivy_podman.py:20]: nitpick: alignment of the ending `#`

File: .pre-commit-config.yaml (hunk 1 above)
Comment (shubhbapna) [reply .pre-commit-config.yaml:48]: Yes pls I just created a PR for it #83

File: scanners/trivy/trivy.py (hunk 2 above)
Comment (jeremychoi) [reply scanners/trivy/trivy.py]: good point. removed this. instead the trivy_cli is initiated in __init__() according to container.paramenters.executable

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jeremychoi) [reply .pre-commit-config.yaml]: reverted

**Comment by jeremychoi** [comment]
@cedricbu pls take another look.

**Comment by jeremychoi** [comment]
we decided to replace this with a new design to support custom scanners more efficiently
